Makefile

```diff
@@ -1,9 +1,8 @@
-CC=$(HOME)/cheri/output/sdk/bin/clang
-CFORMAT=$(HOME)/cheri/output/sdk/bin/clang-format
-CXX=$(HOME)/cheri/output/sdk/bin/clang++
-CFLAGS=-fuse-ld=lld --config cheribsd-riscv64-purecap.cfg
-SSHPORT=10021
-export 
+ifndef ARCH
```

Comment:
I'm pretty sure I'll forget to define `ARCH` and then on risc-v I'll type `make` and be confused at the resulting weird errors! We could either perhaps:

1. use something like `uname` to automatically guess the right architecture
2. have two separate files `Makefile.riscv` and `Makefile.morello` to separate things out in an unambiguous way

I'm fairly neutral between the two.

Reply:
I should SSH into to ``uname`` while cross-compiling. I'll go for the second, then.

Reply:
Should we be cross-compiling on a remote machine? Or should we be saying "actually you need to compile these on a CHERI machine directly"? I would favour the latter (cross compiling always ends in tears!).

Reply:
It is very important to be able to corss-compile imo. As compiling on qemu will take ages.

Reply:
Can we roughly time how long it takes?

Perhaps I've actually over-stated things in one way. I dislike cross-compilation, but I have no problem with making it *possible*. However at the moment I think we *mandate* cross-compilation with the `ssh` stuff?

Reply:
1. There is no clang compiler on CheriBSD by default. So you need to compile and install it additionally.
2. We do not mandate it. The `ssh` stuff is to run when cross compiling.

Reply:
I didn't mean linking SSH and cross-compiling in my comment above :)

Reply:
OK, I'm making this PR go off-piste, which is bad of me. I think there's a deeper issue here that we can come back to in a different PR.